src/DSE.Open.Values/AgeInMonths.cs

```diff
@@ -1,201 +1,184 @@
 // Copyright (c) Down Syndrome Education International and Contributors. All Rights Reserved.
 // Down Syndrome Education International and Contributors licence this file to you under the MIT license.
 
-using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics;
 using System.Runtime.InteropServices;
 using System.Text.Json.Serialization;
 using DSE.Open.Values.Text.Json.Serialization;
 
 namespace DSE.Open.Values;
 
-[JsonConverter(typeof(JsonStringAgeInMonthsConverter))]
+[JsonConverter(typeof(JsonSpanSerializableValueConverter<AgeInMonths, int>))]
+[ComparableValue]
 [StructLayout(LayoutKind.Auto)]
-public readonly record struct AgeInMonths : ISpanFormattable, ISpanParsable<AgeInMonths>, IComparable<AgeInMonths>
+public readonly partial struct AgeInMonths : IComparableValue<AgeInMonths, int>
 {
-    private const string DefaultFormat = "0";
+    private const int MaxYears = int.MaxValue / 12;
+    private const int MaxMonths = int.MaxValue;
 
     public static readonly AgeInMonths Zero;
 
-    public AgeInMonths(int years, int months) : this((years * 12) + months)
+    public static int MaxSerializedCharLength => 12;
+
+    public AgeInMonths(int years, int months)
     {
+        ArgumentOutOfRangeException.ThrowIfNegative(years);
+        ArgumentOutOfRangeException.ThrowIfNegative(months);
```

Comment:
It doesn't make sense to me to allow negative values, so I took a breaking change.

However, I could perhaps be convinced that we could allow `(2, -12)` to be passed, as long as `TotalMonths >= 0`.